README.md

```diff
@@ -23,15 +25,96 @@ Require in `Capfile` to use the default task:
 require 'capistrano/nvm'
 ```
 
-Configurable options:
+The task then uses command maps to run `nvm.sh` when necessary, as controlled by the setting `:nvm_map_bins`.
+
+## Configuration
+
+### Settings
+
+In your capistrano `config/deploy.rb`:
+
+#### Required
+
+- `:nvm_node`
+  - Accepts: String
+  - Default: None
+  - Notes: The version number must be prefixed with a `v` if `:nvm_verb` is set to `:use`. See examples below.
+
+#### Optional
+
+- `:nvm_type`
+  - Accepts: `:user`, `:system`
+  - Default: `:user`
+  - Notes: Used to determine the location of `nvm.sh`, similar to setting `$NVM_HOME` in your shell.
+
+- `:nvm_custom_path`
+  - Accepts: String
+  - Default: Unused
+  - Notes: Setting this effectively overrides `:nvm_type`.
+
+- `:nvm_map_bins`
+  - Accepts: Ruby word array
+  - Default: `%w{node npm yarn}`
+  - Notes: This is a list of commands that will be executed with NVM support by capistrano.
+
+- `:nvm_verb`
+  - Accepts: `:use`, `:install`
+  - Default: `:use`
+  - Notes: Determines whether the deploy invokes `nvm use` or `nvm install`.
+
+## Examples
+
+### Configuring Capistrano
+
+Using a version of NodeJS that is already installed and managed by NVM:
+
+```ruby
+set :nvm_node, 'v8.10' # Version of node.js to use with NVM
+```
+
+Using a version of NodeJS that we want to download and install:
 
 ```ruby
-set :nvm_type, :user # or :system, depends on your nvm setup
-set :nvm_node, 'v0.10.21'
-set :nvm_map_bins, %w{node npm yarn}
+set :nvm_node, 'lts/carbon' # Friendly names work well...
+set :nvm_verb, 'install' # ...but only with the "install" verb
 ```
 
-If your nvm is located in some custom path, you can use `nvm_custom_path` to set it.
+When NVM is installed at the system level:
+
+```ruby
+set :nvm_node, 'lts'
+set :nvm_verb, 'install'
+set :nvm_type, :system
+```
+
+When NVM is installed in a custom location:
+
+```ruby
+set :nvm_node, 'lts'
+set :nvm_verb, 'install'
+set :nvm_custom_path, '/usr/local/src/nvm' # This overrides :nvm_type so we won't bother setting it
+```
+
+### Using a .nvmrc file
+
+#### In your Capistrano code
+
+In `config/deploy.rb`, check for `.nvmrc` and use its value, or a sensible default like v8.10:
+
+```ruby
+set :nvm_node, File.exist?('.nvmrc') ? File.read('.nvmrc').strip : "v8.10"
+```
+
+#### From your release code
+
+In your rakefile where you `execute :npm`, `:node`, or `:yarn`, add this conditional first:
+
+```ruby
+if test "[ -e #{release_path}/.nvmrc ]"
+  download! "#{release_path}/.nvmrc", '.nvmrc'
+  SSHKit.config.default_env[:node_version] = File.read('.nvmrc').strip
+end
+```
```

Comment:
I don't really get what this is about; you download the `.nvmrc` and use that for the next commands? Why should the remote version be leading? 
Or is this something else and do I not understand this?

Reply:
This example is to show specifically how to configure your deployment with the `.nvmrc` file that comes from your release code.

My use case was that our release code contains a `.nvmrc` file, but reading that file and configuring Capistrano with the value from it required me to `download!` it to the machine where the Capistrano code was executing.

I will admit that there's _definitely_ a possibility that I'm simply Doing It Wrong™ here and just am not aware, but this seemed like the right solution to set the nvm version with the one from my deployed code.

Reply:
> My use case was that our release code contains a `.nvmrc` file, but reading that file and configuring Capistrano with the value from it required me to `download!` it to the machine where the Capistrano code was executing.

If you're running the Capistrano commands from the project directory locally, the same `.nvmrc` fill should be already present there, right? So I think your example is only relevant for people with a non-standard setup, and for that I would rather leave it out completely.
Let me know if you think it _is_ common use, or if I didn't understand you correctly.

Reply:
>If you're running the Capistrano commands from the project directory locally, the same .nvmrc fill should be already present there, right? 

In our setup, the Capistrano code runs on a Jenkins server, and the release code ends up on our web nodes.  Capistrano uses SSHKit to run a bunch of `git` commands on each of our web servers in turn to fetch precisely the branch that's been configured for deployment. As a result, the `.nvmrc` file that's in our application code ("release" code in Capistrano parlance) never actually ends up on the Jenkins box where the Capistrano code is running. Hence we download the `.nvmrc`.

I didn't build this deployment and didn't have any experience with Capistrano prior to this, so I just assumed this was normal. Is the typical deployment case for Capistrano different in some way?

The only thing I can see wrong with it is that it's an awfully redundant procedure if you were to have a very large number of nodes. Downloading the same file from all of them is silly but it would still work.